apps/chat/src/store/prompts/prompts.epics.ts

```diff
@@ -332,48 +338,101 @@ const deleteFolderEpic: AppEpic = (action$, state$) =>
+    ),
+    switchMap((promptsListing) => {
+      const foldersIds = Array.from(
+        new Set(promptsListing.map((info) => info.folderId)),
+      );
+      //calculate all folders;
+      const folders = getFoldersFromPaths(
+        Array.from(
+          new Set(foldersIds.flatMap((id) => getAllPathsFromPath(id))),
+        ),
+        FolderType.Prompt,
+      );
+
+      return forkJoin({
+        //get all prompts from api
+        prompts: zip(
+          promptsListing.map((info) => PromptService.getPrompt(info)),
+        ),
+        folders: of(folders),
+      });
+    }),
     tap(({ prompts, folders }) => {
-      //TODO: upload all prompts for export - will be implemented in https://github.com/epam/ai-dial-chat/issues/640
-      exportPrompts(prompts, folders);
+      const filteredPrompts = prompts.filter(Boolean) as Prompt[];
+
+      const appName = SettingsSelectors.selectAppName(state$.value);
+
+      exportPrompts(filteredPrompts, folders, appName);
     }),
     ignoreElements(),
   );
 
 const exportPromptEpic: AppEpic = (action$, state$) =>
   action$.pipe(
     filter(PromptsActions.exportPrompt.match),
-    map(({ payload }) =>
-      PromptsSelectors.selectPrompt(state$.value, payload.promptId),
-    ),
-    filter(Boolean),
-    tap((prompt) => {
-      //TODO: upload all prompts for export - will be implemented in https://github.com/epam/ai-dial-chat/issues/640
+    switchMap(({ payload }) => getOrUploadPrompt(payload, state$.value)),
+
+    switchMap((promptAndPayload) => {
+      const { prompt } = promptAndPayload;
+      if (!prompt) {
+        return of(ImportExportActions.exportFail());
+      }
+
+      const appName = SettingsSelectors.selectAppName(state$.value);
+
       exportPrompt(
         prompt,
         PromptsSelectors.selectParentFolders(state$.value, prompt.folderId),
+        appName,
       );
+      return EMPTY;
     }),
-    ignoreElements(),
   );
 
-const importPromptsEpic: AppEpic = (action$, state$) =>
+const importPromptsEpic: AppEpic = (action$) =>
   action$.pipe(
     filter(PromptsActions.importPrompts.match),
-    map(({ payload }) => {
-      const prompts = PromptsSelectors.selectPrompts(state$.value);
-      const folders = PromptsSelectors.selectFolders(state$.value);
-      //TODO: save in API - will be implemented in https://github.com/epam/ai-dial-chat/issues/640
-      return importPrompts(payload.promptsHistory, {
-        currentFolders: folders,
-        currentPrompts: prompts,
+    switchMap(({ payload }) =>
+      forkJoin({
+        promptsListing: PromptService.getPrompts(undefined, true), //listing of all entities
+        promptsHistory: of(payload.promptsHistory),
+      }),
+    ),
+    switchMap(({ promptsListing, promptsHistory }) => {
+      let foldersIds: (string | undefined)[] = [];
+      let folders: FolderInterface[] = [];
+      let currentPrompts: Observable<(Prompt | null)[]> = of([]);
+
+      if (promptsListing.length) {
```

Comment:
```
if (promptsListing.length === 0) {
    return of({
      currentPrompts: of([]),
      currentFolders: of([]),
      promptsHistory: of(promptsHistory),
    });
  }
  
  const foldersIds = ...;
  const folders = ...;
```
  
  Maybe it's will be better to do something like this?

Reply:
Thank you for suggestions!
Changed.